File: src/main/java/ru/bkmz/util/gui/elements/CustomTableView.java
package ru.bkmz.util.gui.elements;

import javafx.event.EventHandler;
import javafx.scene.Node;
import javafx.scene.control.TableView;
import javafx.scene.input.MouseEvent;
import javafx.scene.layout.Region;

import java.util.Set;

public class CustomTableView<S> extends TableView<S> {
    private final EventHandler<MouseEvent> consumeEvent = MouseEvent::consume;

    @Override
    protected void layoutChildren() {
        super.layoutChildren();
        final Set<Node> dragRects = lookup("TableHeaderRow").lookupAll("Rectangle");
        for (Node dragRect : dragRects) {
            dragRect.removeEventFilter(MouseEvent.ANY, consumeEvent);
            dragRect.addEventFilter(MouseEvent.ANY, consumeEvent);
        }
    }
}

File: src/main/kotlin/ru/bkmz/util/gui/elements/CustomTableView.kt
package ru.bkmz.util.gui.elements

import javafx.event.EventHandler
import javafx.scene.control.TableView
import javafx.scene.input.MouseEvent

class CustomTableView<S> : TableView<S>() {
    private val consumeEvent =
        EventHandler { obj: MouseEvent -> obj.consume() }

    override fun layoutChildren() {
        super.layoutChildren()
        val dragRects = lookup("TableHeaderRow").lookupAll("Rectangle")
        for (dragRect in dragRects) {
            dragRect.removeEventFilter(MouseEvent.ANY, consumeEvent)
            dragRect.addEventFilter(MouseEvent.ANY, consumeEvent)
        }
    }
}